src/Php80/Php80.php

```diff
@@ -57,6 +57,15 @@ public static function get_debug_type($value): string
         return (get_parent_class($class) ?: key(class_implements($class)) ?: 'class').'@anonymous';
     }
 
+    public static function get_resource_id($res): int
+    {
+        if (@\get_resource_type($res) === null) {
+            throw new \TypeError(sprintf('Argument #1 ($res) must be of type resource, %s given', \gettype($res)));
```

Comment:
the message should match the one from PHP8
looking at https://wiki.php.net/rfc/consistent_type_errors, this should be more like
`Argument 1 passed to get_resource_id() must be of the type int, string given`

Reply:
well, `resource` rather than `int` of course :smile: